Docstrings and indentation, bugfix and new method.

danielmader opened the pull request:
I've fixed some code style issues and a potential bug in display_bottom_half(), and added a new clear_all_bytes() to experiment with.
main.py should not be merged (custom pinout)!

danielmader committed:
Docstrings added, indentation fixed.

matrixdata.MatrixData.clear_all_bytes() added.
hub75.Hub75Spi.display_bottom_half() fixed (automatic panel size).

danielmader committed:
Example updated in README.md

benjohnemmett commented on src/matrixdata.py:
```diff
@@ -34,16 +84,55 @@ def set_pixel_value(self, row, col, val):
+        self.red_matrix_data = [bytearray(self.col_bytes) for x in range(self.row_size)]
+        self.green_matrix_data = [bytearray(self.col_bytes) for x in range(self.row_size)]
+        self.blue_matrix_data = [bytearray(self.col_bytes) for x in range(self.row_size)]
+        # self.red_matrix_data = [bytearray(self.col_bytes)] * self.row_size
```
Same with these commented lines?

danielmader replied:
Done!

benjohnemmett replied:
I'm not sure why it wouldn't work, but I suspect it might actually produce a list of references to the same bytearray object instead of creating one bytearray object for each row. It seems like that would still run, but just set every row the same. That's just a guess though, I haven't tried it.

danielmader replied:
This is it, you are absolutely right! If fully explains what I am seeing then. Thank you again!

benjohnemmett commented on src/main.py:
```diff
@@ -8,24 +8,63 @@
 config = hub75.Hub75SpiConfiguration()
+##-----------------------------------------------------------------------------
+## row select pins
+config.line_select_a_pin_number = 15
```
I don't see a way to merge a pull request without merging all of the changes. Could you revert this to the version of main.py on the main branch?

danielmader replied:
Yes, of course, no problem! I'm an absolute beginner with Git(Hub), please excuse.

danielmader replied:
Should be OK now, please check!

benjohnemmett commented:
These changes look great. I appreciate the formatting cleanup and the docstrings.

danielmader commented on src/matrixdata.py:
```diff
@@ -3,29 +3,79 @@
+        self.red_matrix_data = [bytearray(self.col_bytes) for x in range(self.row_size)]
+        self.green_matrix_data = [bytearray(self.col_bytes) for x in range(self.row_size)]
+        self.blue_matrix_data = [bytearray(self.col_bytes) for x in range(self.row_size)]
+        # self.red_matrix_data = [bytearray(self.col_bytes)] * self.row_size
```
Yes, will do! 
Do you haven an explanation for why this does not work in clear_all_bytes()? Only the list comprehension works, but I can't see a difference in the created objects (on CPython).

danielmader replied:
Done!

danielmader committed:
Add files via upload

Reverted back to main.py (with trailing spaces removed).

danielmader committed:
Update matrixdata.py

Removed comments from __init__() and clear_all_bytes():
        # self.red_matrix_data = [bytearray(self.col_bytes)] * self.row_size
        # self.green_matrix_data = [bytearray(self.col_bytes)] * self.row_size
        # self.blue_matrix_data = [bytearray(self.col_bytes)] * self.row_size

danielmader committed:
Update matrixdata.py

Replaced tabs by spaces in 3-Bit Color Mapping docstring.
